File: app/javascript/channels/cursor_channel.js
import consumer from "./consumer"

consumer.subscriptions.create("CursorChannel", {
  connected() {
    console.log('object');
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
  }
});

File: app/channels/cursor_channel.rb
class CursorChannel < ApplicationCable::Channel
  def subscribed
    stream_from "cursor_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def update(data)
    p data
    ActionCable.server.broadcast 'cursor_channel', data
  end
end
